stetho/src/main/java/com/facebook/stetho/inspector/network/ResponseBodyFileManager.java

```diff
@@ -69,6 +89,25 @@ private static String readContentsAsUTF8(InputStream in) throws IOException {
     return out.toString("UTF-8");
   }
 
+  private String prettyPrintContentWithTimeOut(
+      AsyncPrettyPrinter asyncPrettyPrinter,
+      InputStream in) throws IOException {
+    AsyncPrettyPrintingTask prettyPrintingTask = new AsyncPrettyPrintingTask(
+        in,
+        asyncPrettyPrinter);
+    ExecutorService executor = Executors.newSingleThreadExecutor();
+    Future<String> future = executor.submit(prettyPrintingTask);
+    String prettifiedContent = "";
+    try {
+      prettifiedContent = future.get(10, TimeUnit.SECONDS);
+    } catch (TimeoutException | InterruptedException | ExecutionException e) {
+      future.cancel(true);
```

Comment:
This is overly broad.  These 3 exceptions all represent very different problems and should be handled differently.  For starters, `InterruptedException` can occur spuriously and should be resumed.  For this, you will need a new method, `Util#getUninterruptibly` to match the others.

For `TimeoutException`, of course that's our "expected" case and can be handled as you've outlined here but with a slight tweak that you should explain to the caller in the message the reason that it failed (timeout after <n> seconds).

Finally, for `ExecutionException`, your `Callable` should not be throwing a checked exception so you can upgrade any exception that does occur to runtime exception and crash the app.  See `ExceptionUtil` in Stetho for tools to propagate when this happens.